### PR TITLE
[risk=low][RW-12400] Update an enzyme test and continue refactoring resources

### PR DESCRIPTION
**Pull request by jmthibault79**
Followup to #8520

Tested locally by observing and interacting with resource menus on the Analysis Tab, Data Tab, Cohort Review and Home pages.  I did not test Tanagra.

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [x] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/pages/data/tanagra-dev/tanagra-resource-list.tsx [hunk 1]
```diff
@@ -181,7 +181,12 @@ export const TanagraResourceList = fp.flow(
                 {
                   resource: r,
                   workspace,
-                  menu: <ResourceActionsMenu actions={actions(r)} />,
+                  menu: (
+                    <ResourceActionsMenu
+                      actions={actions(r)}
+                      title={`${getTypeString(r)} Action Menu `}
```
Comment (jmthibault79) [inline ui/src/app/pages/data/tanagra-dev/tanagra-resource-list.tsx:187]: this prop is now required

**Review by evrii — APPROVED** [review]
Tested locally by opening the menus on the home page, data tab, and analysis tab.

File: ui/src/app/components/resources/resource-list-action-menu.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  const inactiveBilling =
+    resource.workspaceBillingStatus === BillingStatus.INACTIVE;
+
+  return cond(
+    [isCohort(resource), () => <CohortActionMenu {...props} />],
+    [isCohortReview(resource), () => <CohortReviewActionMenu {...props} />],
+    [isConceptSet(resource), () => <ConceptSetActionMenu {...props} />],
+    [
+      isDataSet(resource),
+      () => <DatasetActionMenu {...{ ...props, inactiveBilling }} />,
```
[the remark this answers is not in the document]
Comment (jmthibault79) [reply ui/src/app/components/resources/resource-list-action-menu.tsx:40]: extra props are generally OK.  This actually is a reversion to yesterday's behavior!  (see render-resource-card.tsx in #8520)

File: ui/src/app/components/resources/resource-list-action-menu.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        <ResourceListActionMenu
+          resource={testDataSet}
+          workspace={workspaceDataStub}
+          existingNameList={[]}
+          onUpdate={async () => {}}
+        />
+      </MemoryRouter>
+    );
+    fireEvent.click(getByTitle('Dataset Action Menu'));
+    expect(getByText('Export to Notebook')).toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (jmthibault79) [reply ui/src/app/components/resources/resource-list-action-menu.spec.tsx:86]: Added comments for WGS - applies to Dataset only

File: ui/src/app/components/resources/resource-list-action-menu.spec.tsx
[the remark this answers is not in the document]
Comment (jmthibault79) [reply ui/src/app/components/resources/resource-list-action-menu.spec.tsx:1]: Now that this is a component, the behavior changes from "returns null" to "fails to render".

I was not initially successful making this work, but I have now added a test back in for this.  thanks.

**Pull request merged**